packages/synapse-react-client/src/components/useImmutableTableQuery.ts

```diff
@@ -159,24 +160,13 @@ export default function useImmutableTableQuery(
           ? queryRequest(getLastQueryRequest())
           : queryRequest,
       )
-      setLastQueryRequest(newQueryRequest)
 
-      if (newQueryRequest.query) {
-        const clonedQueryJson = JSON.stringify(newQueryRequest.query)
-        if (shouldDeepLink) {
-          const encodedQuery = encodeURIComponent(clonedQueryJson)
-          DeepLinkingUtils.updateUrlWithNewSearchParam(
-            'QueryWrapper',
-            componentIndex,
-            encodedQuery,
-          )
-        }
-        if (onQueryChange) {
-          onQueryChange(clonedQueryJson)
-        }
-      }
+      // Remove null/empty array fields
+      lastQueryRequest.query = removeEmptyQueryParams(lastQueryRequest.query)
+
+      setLastQueryRequest(newQueryRequest)
```

Comment:
Remove the empty fields before saving in state. This better handles the scenario where a facet filter is added then removed. Before this change, the new query would contain `selectedFacets: []`, which semantically matches the original query, but the component would treat it as a different query.

For this scenario, this change would treat the latest query exactly like the initial query. This would result in a cache hit, and also allow us to clear the search parameters (see below).

Reply:
Nice catch (most users will benefit from this performance improvement)